### PR TITLE
feat(g-code-testing): Add functionality to run HTTP requests through G-Code Parser

### PR DESCRIPTION
**Pull request by DerekMaggio**
# Overview

Add functionality to run HTTP requests through G-Code Parser 

# Changelog

* Add filtering of __init__.py files in `configurations` command
* Convert ProtocolRunner into an ABC called GCodeEngine
    * In ABC define an abstract `run` method
    * Rename all references
* Create ProtocolGCodeEngine which runs protocols
* Create HTTPGCodeEngine which runs HTTP requests
* Create configuration for `robot/move` HTTP command

# Review requests

Is there something better I can do than `asyncio.run` in g_code_engine and the configuration to actually run the async functions?

# Risk assessment

None

### CODE REVIEW

File: g-code-testing/g_code_parsing/cli.py [hunk 1]
```diff
@@ -55,10 +55,14 @@ def execute(self) -> str:
             left=self.left_pipette_config, right=self.right_pipette_config
         )
         settings = Settings(smoothie=smoothie_settings, host=self.host)
-        file_path = os.path.join(
-            self.CONFIGURATION_DIR_LOCATION, self.configuration_path
-        )
-        with ProtocolRunner(settings).run_protocol(file_path) as program:
+
+        if self.configuration_path.startswith('protocols'):
```
Comment (sfoster1) [inline g-code-testing/g_code_parsing/cli.py:59]: could we make this a command line argument, or a uri protocol or something? it's kind of weird to have this be depending on the name of an intermediate directory.

**Review by sfoster1 — CHANGES_REQUESTED** [review]
One little weirdness but this looks awesome!

File: g-code-testing/g_code_parsing/cli.py (hunk 1 above)
Comment (DerekMaggio) [reply g-code-testing/g_code_parsing/cli.py:59]: If I did a URI protocol would I just use something like a configuration file that maps the URI to the correct file?

**Comment by DerekMaggio** [comment]
Closing this PR and opening a new one with the changes off of edge now that https://github.com/Opentrons/opentrons/pull/8307 is merged

**Pull request closed without merging**